pkg/services/publicdashboards/service/service_test.go

```diff
@@ -1295,35 +1295,57 @@ func assertOldValueIfNull(t *testing.T, expectedValue bool, oldValue bool, nulla
 }
 
 func TestDeletePublicDashboard(t *testing.T) {
+	store := NewFakePublicDashboardStore(t)
+	pubdash := &PublicDashboard{Uid: "2", OrgId: 1, DashboardUid: "uid"}
+	type mockResponse struct {
+		PublicDashboard *PublicDashboard
+		Err             error
+	}
 	testCases := []struct {
 		Name             string
 		AffectedRowsResp int64
 		ExpectedErrResp  error
 		StoreRespErr     error
+		dashboard        *dashboards.Dashboard
+		mockStore        *mockResponse
 	}{
 		{
-			Name:             "Successfully deletes a public dashboards",
+			Name:             "Successfully deletes a public dashboard",
 			AffectedRowsResp: 1,
 			ExpectedErrResp:  nil,
 			StoreRespErr:     nil,
+			dashboard:        &dashboards.Dashboard{UID: "uid", OrgID: 1, IsFolder: false},
+			mockStore:        &mockResponse{pubdash, nil},
 		},
 		{
-			Name:             "Public dashboard not found",
+			Name:             "Public dashboard not found by UID",
 			AffectedRowsResp: 0,
-			ExpectedErrResp:  nil,
+			ExpectedErrResp:  ErrPublicDashboardNotFound,
 			StoreRespErr:     nil,
+			dashboard:        &dashboards.Dashboard{UID: "uid", OrgID: 1, IsFolder: false},
+			mockStore:        &mockResponse{pubdash, ErrDashboardNotFound},
 		},
 		{
-			Name:             "Database error",
+			Name:             "Public dashboard not found by database",
 			AffectedRowsResp: 0,
+			ExpectedErrResp:  ErrInternalServerError,
+			StoreRespErr:     nil,
+			dashboard:        &dashboards.Dashboard{UID: "uid", OrgID: 1, IsFolder: false},
+			mockStore:        &mockResponse{nil, nil},
+		},
+		{
+			Name:             "Failed to delete - Database error",
+			AffectedRowsResp: 1,
 			ExpectedErrResp:  ErrInternalServerError.Errorf("Delete: failed to delete a public dashboard by Uid: uid db error!"),
 			StoreRespErr:     errors.New("db error!"),
+			dashboard:        &dashboards.Dashboard{UID: "uid", OrgID: 1, IsFolder: false},
+			mockStore:        &mockResponse{pubdash, nil},
 		},
 	}
 
 	for _, tt := range testCases {
 		t.Run(tt.Name, func(t *testing.T) {
-			store := NewFakePublicDashboardStore(t)
+			store.On("Find", mock.Anything, mock.Anything).Return(tt.mockStore.PublicDashboard, tt.mockStore.Err)
```

Comment:
Now you are calling 2 methods on the store maybe you need to add another `mockStore` one for Find and another for Delete to mock each case when necessary.

Reply:
I did not understand this. I would need both methods on the same store right? lmk

Reply:
Correct, but you want to mock their responses in a different way. What I'm saying is maybe you need a field `mockFindStore` and another one like `mockDeleteStore`.

Reply:
right, changed accordingly